src/TextShow/Data/VectorNonEmpty.hs

```diff
@@ -0,0 +1,27 @@
+{-# LANGUAGE TemplateHaskell #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+{-|
+Module:      TextShow.Data.Vector.NonEmpty
+Copyright:   (C) 2014-2023 Ryan Scott
+License:     BSD-style (see the file LICENSE)
+Maintainer:  Ryan Scott
+Stability:   Provisional
+Portability: GHC
+
+'TextShow' instances for @NonEmptyVector@ types.
+-}
+module TextShow.Data.VectorNonEmpty () where
+
+import qualified Data.Vector.NonEmpty as B (NonEmptyVector, toList)
+
+import           TextShow (TextShow(..), TextShow1(..), showbPrec1)
+import           TextShow.Utils (showbUnaryListWith)
+
+instance TextShow a => TextShow (B.NonEmptyVector a) where
+    showbPrec = showbPrec1
+    {-# INLINE showbPrec #-}
+
+instance TextShow1 B.NonEmptyVector where
+    liftShowbPrec _ sl p = showbUnaryListWith sl p . B.toList
+    {-# INLINE liftShowbPrec #-}
```

Comment:
This doesn't match how the `Show1` instance for `NonEmptyVector` is implemented upstream, which is (approximately):

```hs
newtype NonEmptyVector a = NonEmptyVector (Vector a)
  deriving newtype Show
```

As such, the `TextShow1` instance for `NonEmptyVector` should be implemented directly in terms of the instance for `Vector`. Something like this ought to work, I suspect:

```hs
import Data.Vector.NonEmpty (toVector)

instance TextShow1 NonEmptyVector where
  liftShowbPrec sp sl p = liftShowbPrec sp sl p . toVector
```